config/swarm-sip.yml

```diff
@@ -49,55 +46,14 @@ kdump_enable: false
-  name: Ben Mort
-  groups: ['wheel','systemd-journal']
-  uid: 1005
-  ssh_key:
-  - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQCeGbpm1SHEdEEquMw+xyrZ0TqULQ1aL4LuDsbF3NQedAxyg+1/PvY3/+HwSs8dZUlP/DGVrgPdP7JEfKi14oF4TQ6oIy89bKkH0QUF5DkIP8g6cHTZZjZxw1Gso47N636u1qw46dSKiBrYFMmEzRMbPVOWGzxdYHP6FSryrl6YiDtiegBA6/j6lM3a0DdUHtV10KhU58NyNl4MSVyQrayqYtlqZK05QFQ47lpE8FHU0Ud6AHB4W4D8Hy0lbUeIVN+S9xZWKRvCmyQSeIX9mLL3rAvDgFYLbGxYCQ4GTnG1U/Ky2CpZBh1n5ihtDINuXDN0uk9YWNMtAGjm4YUqhSopgLKC3EY/fXtZ28Nr3AMtggBD+vY+FnemSFegoTzKwExmTDBEAzTjZK40parklOBQYimgy4iBeD7//iVis/RFKw1vAKLUCPKkx/eSkLhr9e1IqzKcPIWF4hYVRTGMdM42ziJZ/DHiGcOWKRxDzdhM/wT8SZ0xN7H5h8g7pn+SAmWQETLX4I2qvY2qXx0RcoXzB/cNvAOqxtXFTYZZi1nhzZ4u4S6hD1zOalxg6r5CT5VgqL5AHoe3EmFcvD5t0loGWO1e7CqHV0KOCj9jzOTju+BlQbZEPo8zcZfWlpE+s807NkRDqr1Obg2Sbc6kPPtGlNJcxlDMQymHTPnxgH4e9Q== hpcmort1@ilab-gate.ilab.cluster"
-- username: mark
-  name: Mark Goddard
-  groups: ['wheel','systemd-journal']
-  uid: 1006
-  ssh_key:
-  - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCUOhekNNoWCVzdyks2ROdW4gjDxKxPXCMc0VetY08i/WkZPFFUZZOhAN13EVzNh2aVtRYIQSGheqbuIW5QvEIBJ4OqmF5Tjw/76JBQ2c8Vhq0wLboYnvscMD4fwyHbcezBZ67eDxz1P7jTJtoBrshm+vaIXP1PEsQqJOc6RN+a9F6Fqfjuxc+yjGkuwC0h61pdkVrvvqBHUa5cSyFo3A4BAkpzwq785uBSkC/jQH3CxzcbeomxG+1nxYM69qCDARxnPBkHPQHjHP5TxrpS/JS1/pPS6GlKDQ0NvZ+5Eo09QFt7kPoY+05TWwkwCOWXESMPKBFUmOowUgThCZyR/YW3 mark@mark-xps15"
-- username: vlad
-  name: Vlad Stolyarov
-  groups: ['wheel','systemd-journal']
-  uid: 1007
-  ssh_key:
-  - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDeHHJ2Thsnc/F4bx6ZjjKZV/BX68c3Xnl40Aie69wJ00Uh1maYP1p6Pj1QSreFZbYhx11XqwD1a6rlPASAJe/cYj4PIsvmbgAGZqzEpXsWqJij/R7BaYm7EoWiqx4iI7jXCCDE8QJfZC4gR2SPhD3ClLc07SCuvOm6OABJq+KUih0jrkLPcNNK9xiufhe/gP2lfiy7aUPBZThkd6AVgWV+nHaK2dm7d532ChARSpB/Kr3bsyONFDI0+sCxUDtcEHblRw/VVubMLir6UTsYchtCDVVlR3fuiDVM4Oj/p/P8P9OE5VuJ6le1gkYKVK1JFA9LqgPGsbYsuhvylDUe8Acl Generated-by-Nova"
-- username: fred
-  name: Fred Dulwich
-  groups: ['wheel','systemd-journal']
-  uid: 1008
-  ssh_key:
-  - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQClobEi5Eg5GAZk0o43Vb+KgAh6UmTm+rtmaLXVBTZp+e3BubDGK2RVNzmLgXVsu+9P7YdMC65c9E8iC2WXkLmdjTbXYRqCbgiMs6/MCe40YIhV/iBZvyBS8Rb46VKGPBjfJIvU9P2Cs7CYA6Uh0/Vyg1LfFDfq3SfzNBUK5/+gSzR2DyB6hptazq40qa9z/StZ8Qsa0xvIFEV5+eDqPB51MBhZLn3tjy+4tWNRMIgx2CQllYt38w/dtFsLcZvXHEyfm3fYjqfemWDJa9PNMpGB4lSRKSskH4+sVLDGRpCoGp10EXkcfYwjGkNaFA+ZfSpy0Sc0VEPDpdpg+H7tM9dvPHICrJ7ITQlgElKRsUX9H8H/rFsgqf18dWg9rF5Zi4wikSQR30JZbwHZUjhRamq/rJ6LbmfYjMB6tzr9RTyZVT8V8HO9BVvxO+3erIMpSALc6c9wgTxI5BTdhKDInW1uX2OY73RkUhWtCR7QjrpefyS7EI/ya8RvorXcKUxGj1E60Cjcl5EvbuSZk7Z51CKlK2w5uquDamJnqjUCqGzrnkMaIBlVXzfDXVy4VQnAETiC/oeGmx+qQrc3bqkqcg0xADWd9wWvo1rLA83Wh/KGCNkCUK6Wekpi9B+Cv5xUq9GBhGwQ5Pqhzy4WtcOPhrdWPnhf0ZElVL+XovEQKxKq0Q== hpcdulw1@ilab-gate.ilab.cluster"
+# The user is fedora for Kubernetes and Docker Swarm container clusters.
+alaska_admin_user: fedora
 
-# SIP sudoers
-sudoers_sudoers:
-  privileges:
-  - name: "%wheel"
-    entry: "ALL=NOPASSWD:ALL"
+# Extra administrators for the cluster
+extra_admins:
```

Comment:
cluster_users?

Reply:
In reality, cluster_users for container infra do not need to ssh in, they can to coe cluster config to generate the config to driver the cluster, this is in actual fact to give access to additional admin users other than the ones defined in group_vars/all/alaska.

Reply:
Previously the list of `users` added non-admin users. I agree this shouldn't be necessary for a COE, but in practice they wanted to be able to tweak sysctls etc. The users here shouldn't be given admin access.

The line is a bit blurred here - what 'admin' means in this case is the ability to SSH in as the default cloud user (centos or fedora), which is necessary to execute playbooks in p3-appliances. All users are added to the 'wheel' group, so do have root access.

Reply:
I think creating additional users with root access is equivalent to adding SSH keys to the existing admin user. This is not the primary mode we expect users to interact with the cluster as in the case of Slurm. But we do not want to add these SSH keys to every cluster. Hence cluster specific `extra_admin` makes sense to me.